### PR TITLE
fix(lsp): update `inlay_hints.is_enabled` to new nightly format

### PR DESCRIPTION
**Pull request by NicolasGB**
Hi! Thanks for such a great plugin.

Neovim has changed the parameters of the `is_enabled` now it takes a filter similar to a change they did for enabling them a few weeks ago! here's a quick fix, if it suits you!

### CODE REVIEW

File: lua/rustaceanvim/server_status.lua [hunk 1]
```diff
@@ -18,7 +18,7 @@ function M.handler(_, result, ctx, _)
   -- as soon as rust-analyzer has fully initialized.
   if type(vim.lsp.inlay_hint) == 'table' then
     for _, bufnr in ipairs(vim.lsp.get_buffers_by_client_id(ctx.client_id)) do
-      if vim.lsp.inlay_hint.is_enabled(bufnr) then
+      if vim.lsp.inlay_hint.is_enabled({ bufnr = bufnr }) then
```
Comment (mrcjkb) [inline lua/rustaceanvim/server_status.lua:21]: style:

```suggestion
      if vim.lsp.inlay_hint.is_enabled { bufnr = bufnr }  then
```

**Review by mrcjkb — APPROVED** [review]
Hey 👋 

Thanks for the contribution 🙏
I'm away from my laptop now, but I'll push a new release tag later.

File: lua/rustaceanvim/server_status.lua [hunk 2]
```diff
@@ -18,7 +18,7 @@ function M.handler(_, result, ctx, _)
   -- as soon as rust-analyzer has fully initialized.
   if type(vim.lsp.inlay_hint) == 'table' then
     for _, bufnr in ipairs(vim.lsp.get_buffers_by_client_id(ctx.client_id)) do
-      if vim.lsp.inlay_hint.is_enabled(bufnr) then
+      if vim.lsp.inlay_hint.is_enabled { bufnr = bufnr }  then
```
Comment (mrcjkb) [inline lua/rustaceanvim/server_status.lua:21]:
```suggestion
      if vim.lsp.inlay_hint.is_enabled { bufnr = bufnr } then
```

**Comment by mrcjkb** [comment]
CI is failing with unrelated deprecations (after tarheting the `update_flacke_lock_action` branch, which has the updated neovim nightly).
I'll have to fix that later...

**Comment by mrcjkb** [comment]
Merging into the `update_flake_lock_action` branch, and will fix CI there.

**Comment by NicolasGB** [comment]
Thanks! Glad I could help even though it was just a line

**Pull request merged**